content/tutorial/02-advanced-orders/03-pre-signed-orders/01-create-pre-signed-order/README.md

```diff
@@ -0,0 +1,370 @@
+
+```typescript
+/// file: run.ts
+// ...
+
+export async function run(provider: Web3Provider): Promise<unknown> {
+  // ...
+  const settlementContract = new Contract(COW_PROTOCOL_SETTLEMENT_CONTRACT_ADDRESS[chainId], abi)
+  // ...
+}
+```
+
+### Connect to the Safe
+
+Now that we have the `Safe` address and associated helper functions, we can connect to the `Safe`:
+
+```typescript
+/// file: run.ts
+// ...
+
+export async function run(provider: Web3Provider): Promise<unknown> {
+  // ...
+  const { safeApiKit, safeSdk } = await getSafeSdkAndKit(safeAddress)
+  // ...
+}
+```
+
+### Get a quote
+
+As per normal, we are going to request a quote to buy COW tokens with wxDAI using `OrderBookApi`:
+
+```typescript
+/// file: run.ts
+// ...
+import {
+  SupportedChainId,
+  OrderBookApi,
++++  SigningScheme,+++
++++  OrderQuoteRequest,+++
++++  OrderQuoteSideKindSell,+++
++++  OrderCreation,+++
+  COW_PROTOCOL_SETTLEMENT_CONTRACT_ADDRESS
+} from '@cowprotocol/cow-sdk'
+// ...
+
+export function run(provider: Web3Provider): Promise<unknown> {
+  // ...
+  const sellAmount = '1000000000000000000';
+  const sellToken = '0xe91D153E0b41518A2Ce8Dd3D7944Fa863463a97d';
+  const buyToken = '0x177127622c4A00F3d409B75571e12cB3c8973d3c';
+
+  const quoteRequest: OrderQuoteRequest = {
+    sellToken,
+    buyToken,
+    receiver: safeAddress,
+    sellAmountBeforeFee: sellAmount,
+    kind: OrderQuoteSideKindSell.SELL,
+    appData: appDataContent,
+    appDataHash: appDataHex,
+    from: ownerAddress,
+  }
+
+  const { quote } = await orderBookApi.getQuote(quoteRequest);
+  // ...
+}
+```
+
+### Submit the order
+
+Now that we have the quote, we can submit the order to the order book. As we're using `PreSign`, we need to pay special attention to the fields: 
+
+- `from`: the address of the `Safe` wallet
+- `signature`: empty bytes, i.e. `0x`
+- `signingScheme`: `SigningScheme.PRESIGN`
+
+```typescript
+/// file: run.ts
+// ...
++++import { BigNumber, Contract, ethers } from 'ethers'+++
+// ...
+
+export function run(provider: Web3Provider): Promise<unknown> {
+  // ...
+  const order: OrderCreation = {
+    ...quote,
+    sellAmount,
+    buyAmount: BigNumber.from(quote.buyAmount).mul(9950).div(10000).toString(),
+    feeAmount: '0',
+    appData: appDataContent,
+    appDataHash: appDataHex,
+    partiallyFillable: true,
+    +++from: safeAddress,+++
+    +++signature: '0x',+++
+    +++signingScheme: SigningScheme.PRESIGN,+++
+  }
+
+  const orderUid = await orderBookApi.sendOrder(order)
+  // ...
+}
+```
+> The above applies a 0.5% slippage to the order.
+
+> This is the first instance demonstrating the use of a `limit` order. This is done by setting the `feeAmount` to `0`. **Caution**: Ensure that you have set the metadata order class to `limit` otherwise the order will be rejected.
```

Comment:
Will it be rejected really?
I understood backend did not use this metadata at all.

How will this work when fee=0 is live?

AFAICT, now backend will derive the class based on the price being under/over market rate, since all orders will have fee=0.

While it's great to have it match and it helps with our analytics, this is not required.

Reply:
From my understanding, the rejection occurs in the event that `feeAmount != 0` and the user _has specified_ metadata that the order class **is** `limit`. To your point, if _no_ metadata was supplied, and `feeAmount == 0` was specified, it would happily pass through the API.

Reply:
Oh! Actually, I have to check how the `OrderClass` is determined within the `services`, as the `UnsupportedOrderType` API error received is likely due to https://github.com/cowprotocol/services/blob/42266bf72931215dd5cb6a97df771aad1cdb4b2a/crates/shared/src/order_validation.rs#L449 and therefore maybe not implicitly due to the metadata itself.

Reply:
I relaxed the language here and addressed the specific case of partially fillable instead.